### PR TITLE
hooking up unit tests for some of the existing mosaiq helpers

### PR DESCRIPTION
**Pull request by dg1an3**
well actually just two, the get_patient_name/field helpers

### CODE REVIEW

File: lib/pymedphys/tests/mosaiq/test_mock_patient_table.py [hunk 1]
```diff
@@ -1,68 +1,174 @@
...
-    conn = pymssql.connect(msq_server, user=sa_user, password=sa_password)
-    conn.autocommit(True)
-    cursor = conn.cursor()
-    cursor.execute(f"create database {test_db_name}")
-    cursor.close()
 
-    patient_df = pd.DataFrame([("Larry"), ("Moe"), ("Curly")], columns=["FirstName"])
-    patient_df.index = patient_df.index + 10001  # use the index+10001 as the Pat_ID1
+@pytest.fixture(name="create_mock_patients")
+def fixture_create_mock_patients(check_create_test_db):
```
Comment (dg1an3) [inline lib/pymedphys/tests/mosaiq/test_mock_patient_table.py:42]: pylint seems to complain that it doesn't understand what "check_create_test_db" is here -- not sure why...

Comment (SimonBiggs) [reply lib/pymedphys/tests/mosaiq/test_mock_patient_table.py]: I think it was actually just saying that the variable wasn't being accessed within the function. I adjusted it a little to instead call the function within the fixture.

File: lib/pymedphys/tests/mosaiq/test_mock_patient_table.py [hunk 2]
```diff
@@ -1,68 +1,175 @@
...
-@pytest.mark.mosaiqdb
-def test_mock_patient_table():
-    msq_server = "."
-    test_db_name = "MosaiqTest77008"
+sa_user = "sa"
+sa_password = "sqlServerPassw0rd"
+
+# set up a SQLAlchemy engine to be used for populating tables
+connection_str = f"mssql+pymssql://{sa_user}:{sa_password}@{msq_server}/{test_db_name}"
+engine = sqlalchemy.create_engine(connection_str, echo=False)
```
Comment (SimonBiggs) [inline lib/pymedphys/tests/mosaiq/test_mock_patient_table.py:16]: All of these calls up here happen on import, so even if the `--mosaiqdb` flag is not used this code here will still run. Can the engine be made as a fixture instead?

File: lib/pymedphys/tests/mosaiq/test_mock_patient_table.py [hunk 3]
```diff
@@ -1,68 +1,176 @@
...
 
-    conn_str = f"mssql+pymssql://{sa_user}:{sa_password}@{msq_server}/{test_db_name}"
-    engine = sqlalchemy.create_engine(conn_str, echo=False)
-    patient_df.to_sql(
+    # create a single dataframe combining the Patient and Ident tables
+    patient_ident_df = pd.DataFrame(
+        [
+            ("Larry", "Fine", "MR8001"),
+            ("Moe", "Howard", "MR8002"),
+            ("Curly", "Howard", "MR8003"),
```
Comment (SimonBiggs) [inline lib/pymedphys/tests/mosaiq/test_mock_patient_table.py:52]: The three stooges 😄

File: lib/pymedphys/tests/mosaiq/test_mock_patient_table.py [hunk 4]
```diff
@@ -1,68 +1,176 @@
...
-    result_all = pymedphys.mosaiq.execute(
-        connection,
-        """
-        SELECT
-            Pat_Id1,
-            FirstName
-        FROM Patient
-        """,
+        # finally spot check Moe
+        assert moe_patient_name == "HOWARD, Moe"
```
Comment (SimonBiggs) [inline lib/pymedphys/tests/mosaiq/test_mock_patient_table.py:108]: This is beautiful. Absolutely beautiful 🙂

File: lib/pymedphys/tests/mosaiq/test_mock_patient_table.py [hunk 5]
```diff
@@ -1,68 +1,176 @@
...
+    with connect(
+        msq_server,
+        port=1433,
+        database=test_db_name,
+        username=sa_user,
+        password=sa_password,
+    ) as connection:
+
+        # test the get_patient_fields helper function
+        txfield_df = get_patient_fields(connection, "MR8002")
```
Comment (SimonBiggs) [inline lib/pymedphys/tests/mosaiq/test_mock_patient_table.py:165]: 🙂

**Review by SimonBiggs — APPROVED** [review]
@dg1an3 this is wonderful. Thank you!

**Pull request merged**